src/Json.Schema.ToDotNet/ClassGenerator.cs

```diff
@@ -381,6 +402,95 @@ private ConstructorDeclarationSyntax GenerateDefaultConstructor()
                             SuffixedTypeName)));
         }
 
+        /// <summary>
+        /// Generates an initialization statement for each property for which the
+        /// schema specifies a default value.
+        /// </summary>
+        /// <remarks>
+        /// The resulting statements are inserted into the default constructor.
+        /// This ensures that the default values are set even if the object is
+        /// not the result of deserializing a JSON instance document.
+        /// <remarks>
+        /// <returns>
+        /// An array containing one initialization statement for each property
+        /// for which the schema specifies a default value.
+        /// </returns>
+        private ExpressionStatementSyntax[] GenerateDefaultInitializations()
+        {
+            var initializations = new List<ExpressionStatementSyntax>();
+
+            foreach (string propertyName in PropInfoDictionary.GetPropertyNames())
+            {
+                if (IncludeProperty(propertyName))
+                {
+                    PropertyInfo propInfo = PropInfoDictionary[propertyName];
+                    object defaultValue = propInfo.DefaultValue;
+
+                    ExpressionStatementSyntax initializationStatement = GenerateDefaultInitialization(propertyName, defaultValue);
+                    if (initializationStatement != null)
+                    {
+                        initializations.Add(GenerateDefaultInitialization(propertyName, defaultValue));
+                    }
+                }
+            }
+
+            return initializations.ToArray();
```

Comment:
>initializations.ToArray(); [](start = 19, length = 26)

why are you taking the memory costs to generate an array? mostly a List is appropriate for places where an array would be (a List can be enumerated, for example). an Array, in fact, is less flexible than a list, as its size isn't mutable.

Reply:
Because of the shape of the Roslyn API. The result of this method call will be passed to Roslyns `ConstructorDeclarationSyntax.AddBodyStatements`, which takes an argument of type `params StatementSyntax[]`. I'll see if there's a way to add the statements that doesn't require the array. Hang on...

---
In reply to: [244416245](https://github.com/Microsoft/jschema/pull/60#discussion_r244416245) [](ancestors = 244416245)

Reply:
This is a much discussed feature gap in C#. See [https://github.com/dotnet/csharplang/issues/179](https://github.com/dotnet/csharplang/issues/179) and [https://github.com/dotnet/roslyn/issues/36](https://github.com/dotnet/roslyn/issues/36)


---
In reply to: [244418426](https://github.com/Microsoft/jschema/pull/60#discussion_r244418426) [](ancestors = 244418426,244416245)